src/platform/utilities/sso/index.js

```diff
@@ -0,0 +1,51 @@
+import * as Sentry from '@sentry/browser';
+
+import environment from 'platform/utilities/environment';
+import localStorage from '../storage/localStorage';
+import { hasSessionSSO } from '../../user/profile/utilities';
+import mockKeepAlive from './mockKeepAliveSSO';
+import liveKeepAlive from './keepAliveSSO';
+
+const keepAlive = environment.isLocalhost() ? mockKeepAlive : liveKeepAlive;
+let ssoSessionLength = 9000; // milliseconds
+
+export async function ssoKeepAliveSession() {
+  const res = await keepAlive();
+
+  try {
+    const hasSSOsession = res.headers.get('session-alive');
+    if (hasSSOsession === 'true') {
+      // 'session-timeout' is in seconds, convert to milliseconds
+      ssoSessionLength = res.headers.get('session-timeout') * 1000;
+
+      const expirationTime = new Date();
+      expirationTime.setTime(expirationTime.getTime() + ssoSessionLength);
+      localStorage.setItem('sessionExpirationSSO', expirationTime);
+      localStorage.setItem('hasSessionSSO', true);
+    } else {
+      localStorage.removeItem('hasSessionSSO');
+    }
+  } catch (err) {
+    Sentry.withScope(scope => {
+      scope.setExtra('error', err);
+      Sentry.captureMessage(`SSOe error: ${err.message}`);
+    });
+  }
+}
+export function checkAndUpdateSSOeSession() {
+  if (hasSessionSSO()) {
+    const sessionExpiration = localStorage.getItem('sessionExpirationSSO');
+
+    // We want to minimize external keepalive calls, so this functions as self-enforced rate limiting
+    const percentageOfTimeoutThreshold = 0.67;
+    const remainingSessionTime = Date.parse(sessionExpiration) - Date.now();
```

Comment:
I think I'd use moment for this to explicitly set the date format. It might be totally fine but I think `Date.parse` has some browser [concern](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse)

Reply:
This got [mentioned](https://github.com/department-of-veterans-affairs/vets-website/pull/11785#discussion_r389073455) when I originally PR'ed the keep-alive first iteration, and my main concern there was pulling in `moment` for a one-off with performance hits. Is that valid, or this avoided with webpack?